image/descriptor.go

```diff
@@ -66,22 +84,49 @@ func findDescriptor(w walker, name string) (*v1.Descriptor, error) {
 			return err
 		}
 
-		for i := 0; i < len(index.Manifests); i++ {
-			if index.Manifests[i].Annotations[v1.AnnotationRefName] == name {
-				d = index.Manifests[i]
-				return errEOW
+		descs = index.Manifests
+		for _, name := range names {
```

Comment:
The following code is totally duplicate with part code of validate() in image/image.go
Maybe we can create a common or utility function by taking advantage of it.

Reply:
updated, PTAL.